### ISSUE TITLE
Access __version__ is deprecated

### ISSUE THREAD

**Issue by AngellusMortis** [newrelic/newrelic-python-agent#862]
**Description**

```console
/usr/local/lib/python3.10/site-packages/newrelic/common/package_version_utils.py:75: DeprecationWarning: Accessing attr.__version__ is deprecated and will be removed in a future release. Use importlib.metadata directly to query for attrs's 
packaging metadata.
```

**Expected Behavior**

Fix deprecation warning.

**Steps to Reproduce**

1. Import `newrelic` (may want to enable [devmode](https://docs.python.org/3/library/devmode.html))

**Your Environment**

* Python 3.10.11
* newrelic==8.8.1

**Comment by TimPansino** [comment]
We could probably suppress deprecation warnings caused by this function. I'll add that as a feature request.

**Comment by AngellusMortis** [comment]
Suppressing a deprecation is not really a "fix". It will eventually be removed. `importlib-metadata` is part of 3.10 stdlib and it is [backported on PyPi to Python 3.7+](https://pypi.org/project/importlib-metadata/). Migrating to using importlib-metadata as the deprecated suggest seems like the best solution so this does not break in a future version of Python.

**Comment by lrafeei** [comment]
Just to clarify, this is something that will have to be resolved during our "Add support for Python 3.12 before it is released" task in the near future.
